mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -320,14 +341,29 @@ object Word2VecModel extends MLReadable[Word2VecModel] {
     private val className = classOf[Word2VecModel].getName
 
     override def load(path: String): Word2VecModel = {
+      val spark = sparkSession
+      import spark.implicits._
+
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
+
       val dataPath = new Path(path, "data").toString
-      val data = sparkSession.read.parquet(dataPath)
-        .select("wordIndex", "wordVectors")
-        .head()
-      val wordIndex = data.getAs[Map[String, Int]](0)
-      val wordVectors = data.getAs[Seq[Float]](1).toArray
-      val oldModel = new feature.Word2VecModel(wordIndex, wordVectors)
+      val rawData = spark.read.parquet(dataPath)
+
+      val oldModel = if (rawData.columns.contains("wordIndex")) {
```

Comment:
@Krimit You're right that the versioned SaveLoad code was in spark.mllib only.  There isn't a standard to follow yet for spark.ml.  I believe that relying on the Spark version is currently the best option.